app/src/main/java/com/sun/moviedb_48/screen/home/HomeFragment.java

```diff
@@ -24,15 +40,44 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
 
     @Override
     public void initView(View view) {
+        mMovieUpcomingAdapter = new MovieUpcomingAdapter(getContext(), mUpcomingMovies);
+        mUpcomingRecyclerView = view.findViewById(R.id.recyclerViewUpcoming);
+        LinearLayoutManager linearLayoutManager =
+                new LinearLayoutManager(getContext(), RecyclerView.HORIZONTAL, false);
+        mUpcomingRecyclerView.setLayoutManager(linearLayoutManager);
+        mUpcomingRecyclerView.setHasFixedSize(false);
+        DividerItemDecoration dividerItemDecoration = new DividerItemDecoration(getContext(), RecyclerView.HORIZONTAL);
+        mUpcomingRecyclerView.addItemDecoration(dividerItemDecoration);
+        mUpcomingRecyclerView.setAdapter(mMovieUpcomingAdapter);
     }
 
     @Override
     public void initData() {
-
+        MovieRemoteDataSource movieRemoteDataSource = MovieRemoteDataSource.getInstance();
+        MovieLocalDataSource movieLocalDataSource = MovieLocalDataSource.getInstance();
+        MovieRepository movieRepository =
+                MovieRepository.getInstance(movieRemoteDataSource, movieLocalDataSource);
+        mPresenter = new HomePresenter(movieRepository);
+        mPresenter.setView(this);
+        mPresenter.getMoviesByCategories(Categories.UPCOMING);
     }
 
     @Override
     public void initListener() {
 
     }
+
+    @Override
+    public void onGetMovieByCategoriesSuccess(ArrayList<Movie> movies,
```

Comment:
Mình nghĩ là View thì chỉ đảm nhiệm việc showData, còn việc load data thuộc về presenter, nên cảm thấy tên như này chưa được hợp lý lắm 
```suggestion
    public void showMovies(ArrayList<Movie> movies,
```